prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -86,14 +90,28 @@ async def _arun(
 
 
 class CrewAIAgentSubquestions:
-    def __init__(self, model: str) -> None:
+    def __init__(self, model: str, memory: bool = True) -> None:
         self.model = model
         self.subgraph_handler = OmenSubgraphHandler()
         self.pinecone_handler = PineconeHandler()
+        self.memory = memory
+        if self.memory:
+            self._long_term_memory = LongTermMemory("think-thoroughly-agent")
 
     def _get_current_date(self) -> str:
         return utcnow().strftime("%Y-%m-%d")
 
+    def save_answer_to_long_term_memory(
+        self, answer_with_scenario: AnswerWithScenario
+    ) -> None:
+        if not self._long_term_memory:
```

Comment:
This will throw error if memory is False:

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/038b7080-744b-4c3d-9ca7-d538f4470de0)

I left two suggestions above that will fix it.